chapter02/worksheets/solution2.5.2.8_stream_iterate.scala

```diff
@@ -0,0 +1,47 @@
+/** 
+  Exercise 2.5.2.8
+
+  (a) Reverse an integer’s digits (see Example 2.5.1.6) as shown:
+  def revDigits(n: Int): Int = ???
+
+  scala> revDigits(12345)
+  res0: Int = 54321
+
+  (b) A palindrome integer is an integer number n such that revDigits(n) == n.
+
+  Write a predicate function of type Int => Boolean that checks
+  whether a given positive integer is a palindrome.
+  */
+
+def revDigits(n: Int): Int = {
+  Stream.iterate(n)(_ / 10).takeWhile(_ != 0).map(_ % 10).toList.reverse.foldRight(0)(_ + _ * 10)
+}
+
+val result = revDigits(12345)
+val expected: Int = 54321
+assert(result == expected) 
+
+def isPalindrome(n: Int): Boolean = {
+  n.toString.reverse.toInt == n
```

Comment:
Here, I would expect you to call `revDigits` to get the reversed digits, rather than `toString`.

Reply:
Okay, I'm on it.